deployment/ccip/changeset/v1_6_2/cs_deploy_cctp_message_transmitter_proxy_test.go

```diff
@@ -0,0 +1,195 @@
+package v1_6_2_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/latest/mock_usdc_token_messenger"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/latest/mock_usdc_token_transmitter"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	cldf_chain "github.com/smartcontractkit/chainlink-deployments-framework/chain"
+	cldf_evm "github.com/smartcontractkit/chainlink-deployments-framework/chain/evm"
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	"github.com/smartcontractkit/chainlink-evm/gethwrappers/shared/generated/burn_mint_erc677"
+	"github.com/smartcontractkit/chainlink-evm/pkg/utils"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset/v1_6_2"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/stateview"
+	commoncs "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+)
+
+func setupCCTPMsgTransmitterProxyEnvironmentForDeploy(t *testing.T, withPrereqs bool) (cldf.Environment, []uint64) {
+	env := memory.NewMemoryEnvironment(t,
+		logger.Test(t),
+		zapcore.InfoLevel,
+		memory.MemoryEnvironmentConfig{Chains: 2},
+	)
+
+	selectors := env.BlockChains.ListChainSelectors(cldf_chain.WithFamily(chain_selectors.FamilyEVM))
+	if withPrereqs {
+		var err error
+
+		prereqCfg := make([]changeset.DeployPrerequisiteConfigPerChain, len(selectors))
+		for i, selector := range selectors {
+			prereqCfg[i] = changeset.DeployPrerequisiteConfigPerChain{
+				ChainSelector: selector,
+			}
+		}
+
+		env, err = commoncs.Apply(t, env,
+			commoncs.Configure(
+				cldf.CreateLegacyChangeSet(changeset.DeployPrerequisitesChangeset),
+				changeset.DeployPrerequisiteConfig{
+					Configs: prereqCfg,
+				},
+			),
+		)
+		require.NoError(t, err)
+	}
+
+	return env, selectors
+}
+
+func setupCCTPMsgTransmitterProxyContractsForDeploy(
+	t *testing.T,
+	logger logger.Logger,
+	chain cldf_evm.Chain,
+	addressBook cldf.AddressBook,
+) *cldf.ContractDeploy[*mock_usdc_token_messenger.MockE2EUSDCTokenMessenger] {
+	usdcToken, err := cldf.DeployContract(logger, chain, addressBook,
```

Comment:
Instead of cldf.DeployContract, we should use opsutil.NewEVMDeployOperation now. & then save returned addresses using the datastore API.